src/sentry/digests/utilities.py

```diff
@@ -23,3 +24,117 @@ def get_digest_metadata(digest):
                     end = record.datetime
 
     return start, end, counts
+
+
+def get_personalized_digests(project_id, digest, user_ids):
+    """
+    get_personalized_digests(project: Project, digest: Digest, user_ids: Set[Int]) -> Iterator[user_id: Int, digest: Digest]
+    """
+    # TODO(LB): I Know this is inefficent.
+    # In the case that ProjectOwnership does exist, I do the same query twice.
+    # Once with this statement and again with the call to ProjectOwnership.get_actors()
+    # Will follow up with another PR to reduce the number of queries.
+    if ProjectOwnership.objects.filter(project_id=project_id).exists():
+        events = get_events_from_digest(digest)
+        events_by_actor = build_events_by_actor(project_id, events)
+        events_by_user = convert_actors_to_users(events_by_actor, user_ids)
+        for user_id in six.iterkeys(events_by_user):
```

Comment:
You can switch to `iteritems` here to avoid addressing back into the mapping on the line below.